src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/EmbeddedLanguages/Common/EmbeddedSyntaxToken.cs

```diff
@@ -69,5 +71,36 @@ public EmbeddedSyntaxToken<TSyntaxKind> WithDiagnostics(ImmutableArray<EmbeddedD
 
         public TextSpan GetSpan()
             => EmbeddedSyntaxHelpers.GetSpan(this.VirtualChars);
+
+        public override string ToString()
+            => ToString(skipTrivia: false);
```

Comment:
so roslyn compiler semantics are that we have ToString and ToFullString.  ToString doesn't include leading trivia of the first token or trailing tricia of the last token.  ToFullString includes them.  if you want these mthods on these core types, it should match those semantics.

if you do want a: give me this with *no* trivia, that should live as a helper elsewhere (potentially juse in the stackframe codebase).